app/views/layouts/_header.html.erb

```diff
@@ -17,19 +17,17 @@
       <li> <%= link_to "Cart", cart_path %> </li>
 
       <% if current_user %>
-        <% if current_admin? %>
+        <% if store_owner? %>
           <li><%= link_to "Dashboard", admin_store_path(current_user.store.slug) %></li>
-        <% else %>
-          <li><%= link_to "Profile", profile_path %></li>
         <% end %>
 
+        <li><%= link_to "Profile", profile_path %></li>
```

Comment:
Also, thanks for catching that Profile didn't show in the navbar for an admin. I kept meaning to mention it or deal with it. I'm glad you got it corrected.